qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -368,7 +398,18 @@ private static void validateSectionTag(Section section, Template template,
 	private static boolean canChangeContext(Section section) {
 		SectionKind sectionKind = section.getSectionKind();
 		return sectionKind == SectionKind.EACH || sectionKind == SectionKind.FOR || sectionKind == SectionKind.LET
-				|| sectionKind == SectionKind.SET || sectionKind == SectionKind.WITH;
+				|| sectionKind == SectionKind.SET || sectionKind == SectionKind.WITH
+				|| sectionKind == SectionKind.SWITCH || sectionKind == SectionKind.WHEN;
+	}
+
+	private static boolean isMatchSwitchWhenSection(Section parentSection, Section childSection) {
+		if (parentSection == null || childSection == null) {
+			return false;
+		}
+		SectionKind childSectionKind = childSection.getSectionKind();
+		SectionKind parentSectionKind = parentSection.getSectionKind();
+		return ((parentSectionKind == SectionKind.SWITCH && childSectionKind == SectionKind.CASE)

```

Comment:
Are sure that we should check that #case works only with #switch. I tthink that you  an use #case with #when
I suggest that yiu try that by rendering à qute template with this usecase

Reply:
After playing with Qute template #switch is the sam ething than #when and #is is the same thing than #case, so you can mix them. For instance this template is valid:

```
{#switch "A"}
	{#is "A"}
		A
	{#case "B"}
		B
{/switch}
```

The test chould be just: `sectionKind == SectionKind.SWITCH || sectionKind == SectionKind.WHEN` and you don't need childSection.